src/ngx_http_lua_uthread.c

```diff
@@ -273,6 +273,15 @@ ngx_http_lua_uthread_kill(lua_State *L)
         ngx_http_lua_del_thread(r, L, ctx, sub_coctx);
         ctx->uthreads--;
 
+        /* notify the parent that the thread was killed */
+        parent_coctx = sub_coctx->parent_co_ctx;
+        if (sub_coctx->waited_by_parent && parent_coctx) {
+            parent_co = parent_coctx->co;
+            lua_pushboolean(parent_co, 0);
+            lua_pushstring(parent_co, "user thread killed");
+            ngx_http_lua_post_thread(r, ctx, parent_coctx, 2);
```

Comment:
@aviramc Also, maybe we should avoid the overhead of posting the parent thread if the killer is indeed the parent.